src/app/modules/feature-flags/feature-flags.service.ts

```diff
@@ -23,3 +24,49 @@ export const getEnabledFlags = (): ResultAsync<string[], DatabaseError> => {
     okAsync(enabledFlagsDocs.map((doc) => doc.name)),
   )
 }
+
+/**
+ * Wrapper over getEnabledFlags function to gracefully handle errors
+ * and returns the `options.fallbackValue` instead.
+ *
+ * @example
+ * ```
+ * const logMeta = {
+ *  action: 'linkStripeAccountToForm',
+ * }
+ * return getFeatureFlag(featureFlags.validateStripeEmailDomain, {
+ *  fallbackValue: false,
+ *  logMeta,
+ * })
+ *.andThen((shouldValidateStripeEmailDomain) => ...)
+ *```
+ *
+ * @param flag
+ * @param options.fallbackValue the value to fall back to in the event that there's an error retrieving the feature flag
+ * @returns boolean that represents the status of the feature flag or the fallback value
+ */
+export const getFeatureFlag = (
+  flag: keyof typeof featureFlags,
+  options?: {
+    fallbackValue?: boolean
+    logMeta?: CustomLoggerParams['meta']
+  },
+): ResultAsync<boolean, DatabaseError> => {
+  const _fallbackValue = options?.fallbackValue ?? false
+  return getEnabledFlags()
+    .andThen((featureFlagsListResult) =>
+      okAsync(featureFlagsListResult.includes(flag)),
+    )
+    .orElse((error) => {
+      logger.error({
+        message: 'Error occurred whilst retrieving enabled feature flags',
+        meta: options?.logMeta
+          ? options.logMeta
+          : {
+              action: 'getFeatureFlag',
+            },
+        error,
+      })
+      return okAsync(_fallbackValue)
+    })
+}
```

Comment:
I think its better to incorporate the default `false` into the function typing definition, so something like:
```suggestion
export const getFeatureFlag = (
  flag: keyof typeof featureFlags,
  options: {
    fallbackValue: boolean
    logMeta?: CustomLoggerParams['meta']
  } = { fallbackValue: false },
): ResultAsync<boolean, DatabaseError> => {
  return getEnabledFlags()
    .andThen((featureFlagsListResult) =>
      okAsync(featureFlagsListResult.includes(flag)),
    )
    .orElse((error) => {
      logger.error({
        message: 'Error occurred whilst retrieving enabled feature flags',
        meta: options?.logMeta
          ? options.logMeta
          : {
              action: 'getFeatureFlag',
            },
        error,
      })
      return okAsync(options.fallbackValue)
    })
}
```

Reply:
That changes the option argument in two ways.
1. The default `fallbackValue` will not be set if `options` was passed in from the caller. 
2. `fallbackValue` becomes a required value if I want to override the `logMeta` which may not be the case.

```ts
const options = { logMeta: { foo: "bar" }} 
getFeatureFlag('flag', options) // ts-error fallbackValue is required in "options"
```

Reply:
I see, is there any other way to type `false` as a default value without these side effects then 🤔

Reply:
Sorry for getting back late, was occupied by other urgent tasks.

I've seen different ways of providing a default but there's no clear winner, and each have their own pros and cons.

1. The way that you're seeing now
2. `Object.assign`

Reply:
No worries, okays let's keep it as it is then, thanks Ken!